### ISSUE TITLE
The accuracy of ISystemClock

### ISSUE THREAD

**Issue by BrianAlk** [DuendeSoftware/Support#101]
Hi,

We are using the identity server package a lot and noticed during the logging of events from the DefaultEventService that the accuracy of the timestamp is without milliseconds. This due to the use of the ISystemClock opposed to the more common DateTime objects. 

Are there any future plans to convert the use of ISystemClock to DateTime with the support of milliseconds? This would make the logging more accurate. 

Thanks in advance

**Comment by BrianAlk** [comment]
Sounds like a fair solution. I understand that it is a breaking change. Especially since I don't see Microsoft changing their SystemClock any time soon.

**Comment by brockallen** [comment]
I guess the other angle is to use something else in the events system.

**Comment by BrianAlk** [comment]
Correct me if I am wrong, but theoretically this line:
`evt.TimeStamp = Clock.UtcNow.UtcDateTime;`
https://github.com/DuendeSoftware/IdentityServer/blob/76dda3a66f3242dbfbb9bd7be78cd48e99fe3bdf/src/IdentityServer/Services/Default/DefaultEventService.cs#L116

Will result in a DateTime regardless of the interface used. So potentially even only updating this instance would work.

**Comment by brockallen** [comment]
> Will result in a DateTime regardless of the interface used. So potentially even only updating this instance would work.

Yes, this could be a short-term fix so you get your logs with timestamps.

**Comment by brockallen** [comment]
I submitted a PR for the short term fix (please have a look) and opened an issue to consider a longer term approach. I'll close this issue now. Thanks.

**Comment by BrianAlk** [comment]
Much appreciated, thanks!
